cql3/expr/expression.hh

```diff
@@ -131,6 +131,26 @@ extern value_set possible_lhs_values(const column_definition*, const expression&
 /// Turns value_set into a range, unless it's a multi-valued list (in which case this throws).
 extern nonwrapping_range<bytes> to_range(const value_set&);
 
+/// A range of all X such that X op val.
+template<typename T>
+nonwrapping_range<T> to_range(oper_t op, const T& val) {
```

Comment:
Perhaps keep it out-of-line an explicitly instantiate it for the types we need (there are probably just one or two) with `extern template`. See #1.

Reply:
> Perhaps keep it out-of-line an explicitly instantiate it for the types we need (there are probably just one or two)

This part I understand and have done.

> with `extern template`.

This part confuses me.  I thought `extern template` only applies if I keep the inline definition?

Reply:
Yes, my mistake.